src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/GlobalJsonReader.cs

```diff
@@ -59,86 +55,88 @@ internal static class GlobalJsonReader
             }
         }
 
-        /// <summary>
-        /// Deserializes a global.json and returns the MSBuild SDK versions
-        /// </summary>
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private static Dictionary<string, string> Deserialize(string value)
-        {
-            return JsonConvert.DeserializeObject<GlobalJsonFile>(value).MSBuildSdks;
-        }
-
-        private sealed class GlobalJsonFile
-        {
-            [JsonProperty(MSBuildSdksPropertyName)]
-            public Dictionary<string, string> MSBuildSdks { get; set; }
-        }
-
         /// <summary>
         /// Searches for a file based on the specified starting directory.
         /// </summary>
-        /// <param name="file">The file to search for.</param>
-        /// <param name="startingDirectory">An optional directory to start the search in.  The default location is the directory
-        /// of the file containing the property function.</param>
-        /// <returns>The full path of the file if it is found, otherwise an empty string.</returns>
-        private static string GetPathOfFileAbove(string file, string startingDirectory)
+        /// <param name="fileName">The name of the file to search for.</param>
+        /// <param name="startingDirectory">An optional <see cref="DirectoryInfo" /> representing the directory to start the search in.</param>
+        /// <param name="result">Receives the <see cref="FileInfo" /> of the file if found, otherwise <code>null</code>.</param>
+        /// <returns><code>true</code> if a file was found in the directory or any parent directory, otherwise <code>false</code>.</returns>
+        [System.Runtime.CompilerServices.MethodImpl(System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]
```

Comment:
Is this beneficial? Didn't you fix an issue where the runtime was inlining a method, causing a DLL to be loaded  despite the method never being executed? In other words, it already seems aggressive enough

Reply:
I'm not sure why I put this here, must have been a copy/paste error.  There's no point in not inlining this method since it should only be pulling in types from `System.IO` which is already loaded probably.